torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1858,6 +1871,35 @@ def _lazy_init(self) -> None:
         for handle in self._handles:
             self._init_param_attributes(handle)
         self._exec_order_data.init(self, self.process_group)
+        # Initialize inter-node process group for hybrid shard comm. We assume
+        # passed in process group is the process group over which the model is
+        # sharded, and initialize the iter-node PG here for shard replication.
+        if self.sharding_strategy == ShardingStrategy.HYBRID_SHARD:
+            assert self._is_root
+            sharding_backend = dist.get_backend(self.process_group)
+            # TODO - we are using the overall world size here, but user may wish
+            # to tune this.
+            world_size = dist.get_world_size()
```

Comment:
Does this assume that FSDP's `self.process_group` is the default process group? (I.e. do we need to call `world_size = dist.get_world_size(self.process_group)`?)

Reply:
This is a great catch, we should definitely pass in the process group